### PR TITLE
Add telemetry for creating new windows

### LINKED ISSUE #2347: Add new telemetry for tracking window is opened.
**Is your feature request related to a problem? Please describe.**
- What's the answer you want to get from this telemetry? Please describe it.
We wanna collect the data about how many windows users create in a session.
- Is it measured by time, counter, or a string?
By Counter. Maybe Label counter.
- When will the telemetry metric be sent? Daily? In the end of a session? or something else?
In the end of a session.

### PR DESCRIPTION
**Pull request by daoshengmu**
Fixes #2347

### CODE REVIEW

**Comment by daoshengmu** [comment]
# Request for data collection review form
 **All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**
 
 1. **What questions will you answer with this data?**
 
 *  Counting how many general windows are opened in a session.
 
 2. **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**

 * We also are interested in analyzing users' behavior during a session about how they use FxR multi-window feature.
  
 3. **What alternative methods did you consider to answer these questions? Why were they not sufficient?**
 
 * N/A
 
 4. **Can current instrumentation answer these questions?**
 
 * Currently no, as these are some of the first metrics we're recording.
 
 5. **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection](https://wiki.mozilla.org/Firefox/Data_Collection) [categories](https://wiki.mozilla.org/Firefox/Data_Collection#Data_Collection_Categories) on the found on the Mozilla wiki.**
 
 * All data is Category 2.
 
 6. **How long will this data be collected?**
 
 Until 05/01/2020
 
 7. **What populations will you measure?**

 * All release, beta, and nightly users with telemetry enabled.
 
 8. **Please provide a general description of how you will analyze this data.**
 
 * Glean
 
 9. **Where do you intend to share the results of your analysis?**
 
 * Only on Glean and with Firefox Reality team..

**Comment by daoshengmu** [comment]
@harraton Please help data review.

The metrics document is here, https://github.com/MozillaReality/FirefoxReality/blob/d296121500f2fb386167368539aa99210d7465f8/docs/metrics.md, and it will be shown in docs/metrics.md once we merge it.

The wiki page is here, https://github.com/MozillaReality/FirefoxReality/wiki/Telemetry#new-telemetry-system---glean

**Comment by daoshengmu** [comment]
@Dexterp37 Please help review our Glean usage.

File: app/metrics.yaml [hunk 1]
```diff
@@ -83,3 +83,18 @@ searches:
...
+  open_new_window:
+    type: counter
+    description: >
+      Counting how many general windows are opened in a session.
+    send_in_pings:
+      - session_end
+    bugs:
+      - https://github.com/MozillaReality/FirefoxReality/issues/2347
+    data_reviews:
+      - https://github.com/MozillaReality/FirefoxReality/pull/1854#issuecomment-546214568
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please don't forget to update this once the data-review is granted :)

**Comment by harraton** [comment]
Review passed.

# Data Review Form 
1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 
Yes, according to the comment above: The metrics document is here, https://github.com/MozillaReality/FirefoxReality/blob/d296121500f2fb386167368539aa99210d7465f8/docs/metrics.md, and it will be shown in docs/metrics.md once we merge it.

The wiki page is here, https://github.com/MozillaReality/FirefoxReality/wiki/Telemetry#new-telemetry-system---glean

2) Is there a control mechanism that allows the user to turn the data collection on and off? 
Yes, via the Settings page for Firefox Reality

3) If the request is for permanent data collection, is there someone who will monitor the data over time? Yes, @daoshengmu 

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2

5) Is the data collection request for default-on or default-off?
Default On

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?
No

7) Is the data collection covered by the existing Firefox privacy notice? 
Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) Yes

9) Does the data collection use a third-party collection tool?
Glean.

**Comment by daoshengmu** [comment]
@keianhzo It is data reviewed. Please help review the rest part. Thanks.

File: app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java [hunk 2]
```diff
@@ -249,6 +249,11 @@ public WindowWidget addWindow() {
         focusWindow(newWindow);
         updateCurvedMode(true);
         updateViews();
+
+        // We are only interested in general windows opened.
+        if (!isInPrivateMode()) {
```
Comment (keianhzo) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java]: I guess this has been a conscious decision but we are sending open window events for private windows too in the old telemetry.

Comment (Dexterp37) [reply app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java]: This is interesting. @daoshengmu was this clearly called out and discussed with Data Stewards?

Comment (daoshengmu) [reply app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/Windows.java]: I got this request from here, https://github.com/MozillaReality/FirefoxReality/issues/2347#issuecomment-558807732

@jvonitter Can you confirm it again? If so, we will only collect `open new window` events for non-private mode.

**Pull request merged**